iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestConflictingDataFiles.java

```diff
@@ -0,0 +1,254 @@
+        Assert.fail();
+      }
+    }
+
+    List<Object[]> objects =
+        shell.executeStatement("SELECT * FROM customers ORDER BY customer_id, last_name, first_name");
+    Assert.assertEquals(12, objects.size());
+    List<Record> expected = TestHelper.RecordsBuilder.newInstance(HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA)
+        .add(1L, "Joanna", "Pierce")
+        .add(1L, "Changed", "Taylor")
+        .add(2L, "Changed", "Donnel")
+        .add(2L, "Susan", "Morrison")
+        .add(2L, "Bob", "Silver")
+        .add(2L, "Joanna", "Silver")
+        .add(3L, "Marci", "Barna")
+        .add(3L, "Blake", "Burr")
+        .add(3L, "Trudy", "Henderson")
+        .add(3L, "Trudy", "Johnson")
+        .add(4L, "Laci", "Zold")
+        .add(5L, "Peti", "Rozsaszin").build();
+    HiveIcebergTestUtils.validateData(expected,
+        HiveIcebergTestUtils.valueForRow(HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA, objects), 0);
+
+  }
+
+  @Test
+  public void testMultiFiltersUpdate() {
+
+    PartitionSpec spec =
+        PartitionSpec.builderFor(HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA).identity("last_name")
+            .bucket("customer_id", 16).build();
+
+    // create and insert an initial batch of records
+    testTables.createTable(shell, "customers", HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA, spec, fileFormat,
+        HiveIcebergStorageHandlerTestUtils.OTHER_CUSTOMER_RECORDS_2, 2);
+    // insert one more batch so that we have multiple data files within the same partition
+    shell.executeStatement(testTables.getInsertQuery(HiveIcebergStorageHandlerTestUtils.OTHER_CUSTOMER_RECORDS_1,
+        TableIdentifier.of("default", "customers"), false));
+
+    String[] multiFilterQuery = new String[] {
+        "UPDATE customers SET first_name='Changed' WHERE  last_name='Henderson' " + "OR last_name='Johnson'",
+        "UPDATE customers SET first_name='Changed' WHERE  last_name='Taylor' AND customer_id=1" };
+
+    try {
+      Tasks.range(2).executeWith(Executors.newFixedThreadPool(2)).run(i -> {
+        init(shell, testTables, temp, executionEngine);
+        HiveConf.setBoolVar(shell.getHiveConf(), HiveConf.ConfVars.HIVE_VECTORIZATION_ENABLED, isVectorized);
+        HiveConf.setVar(shell.getHiveConf(), HiveConf.ConfVars.HIVEFETCHTASKCONVERSION, "none");
+        HiveConf.setVar(shell.getHiveConf(), HiveConf.ConfVars.HIVE_QUERY_REEXECUTION_STRATEGIES,
+            RETRY_STRATEGIES_WITHOUT_WRITE_CONFLICT);
+        shell.executeStatement(multiFilterQuery[i]);
+        shell.closeSession();
+      });
+    } catch (Throwable ex) {
+      // If retry succeeds then it should not throw an ValidationException.
+      Throwable cause = Throwables.getRootCause(ex);
+      if (cause instanceof ValidationException && cause.getMessage().matches("^Found.*conflicting.*files(.*)")) {
+        Assert.fail();
+      }
```

Comment:
You are ignoring all other exceptions here, No need of catch I believe let the exception surface